tables/__init__.py

```diff
@@ -8,25 +8,19 @@
 """
 import os
 from ctypes import cdll
-import platform
-
-
-# Load the blosc2 library, and if not found in standard locations,
-# try this directory (it should be automatically copied in setup.py).
-current_dir = os.path.dirname(__file__)
-platform_system = platform.system()
-blosc2_lib = "libblosc2"
-if platform_system == "Linux":
-    blosc2_lib += ".so"
-elif platform_system == "Darwin":
-    blosc2_lib += ".dylib"
-else:
-    blosc2_lib += ".dll"
-try:
-    cdll.LoadLibrary(blosc2_lib)
-except OSError:
-    cdll.LoadLibrary(os.path.join(current_dir, blosc2_lib))
+from importlib.metadata import files as import_metadata_files
 
+try:
+    # if blosc2 is not installed, move on
+    import blosc2
+    for f in import_metadata_files('blosc2'):
+        if f.name in {"libblosc2.dll", "libblosc2.so", "libblosc2.dylib"}:
+            cdll.LoadLibrary(str(f.locate().resolve()))
+            break
+    else:
+        raise FileNotFoundError('Unable to locate the blosc2 binary')
```

Comment:
I think that this is not OK for the cases (like linux distributions) where libblosc2 is typically provided in a standalone package and both python-blosc2 and pytables use the system libraries.
Is there a way to workaround?

Reply:
The issue only arises if blosc2 is installed but it uses the system `libblosc2` because then the import will work but `LoadLibrary` will fail. Could I just remove the `FileNotFoundError` exception and not raise an exception at all?

What would be the downside of that? Eventually an error will be raised by python when it tries to load the blosc2 and can't find it, right? In that cause there's no real reason to raise an error here!?

Reply:
Yes, to me it is fine.

Reply:
Note that linux distributions have a libblosc2.so.2 but not necessarlily a libblosc2.so when there are no developments packages installed.